DEVOPS-693: migrate from conda-lock to pixi

saicheranb opened the pull request:
**DEVOPS-693 - migrate from conda-lock to pixi**

saicheranb committed:
DEVOPS-693 :: Refactor pyproject.toml to streamline dependencies and add pixi configuration

- Removed dynamic dependencies for 'dependencies' in pyproject.toml.
- Added explicit dependencies for numpy, properties, pypng, six, and vectormath.
- Introduced pixi project configuration with channels and platform specifications.
- Defined dependencies for different features (test, dev, doc) under pixi.
- Updated versioning and removed commented-out geoh5py dependency.
- Adjusted test tasks and local dependency installation commands.

Refactor version_test.py to improve readability and efficiency

- Simplified get_conda_recipe_version function by removing unnecessary template rendering.
- Streamlined file reading process for recipe.yaml.

saicheranb committed:
Refactor pyproject.toml: Consolidate Sphinx dependency and remove redundant section

- Added Sphinx dependency to the main dependencies list.
- Removed the separate documentation dependencies section.

saicheranb committed:
Refactor get_conda_recipe_version function to read and render YAML template content

saicheranb committed:
Refactor get_conda_recipe_version function to improve readability and streamline YAML loading

saicheranb committed:
Add Sphinx documentation dependencies and update environment configurations

saicheranb committed:
Update Python version requirement and Sphinx dependency constraints

- Changed the required Python version from ">=3.10,<4.0" to ">=3.9,<4.0" in pyproject.toml to support Python 3.9.
- Updated Sphinx dependency from "*" to ">=5.3,<6.0" for better compatibility and to avoid potential breaking changes in future versions.

saicheranb committed:
Add Sphinx documentation build task to pyproject.toml

saicheranb committed:
Fix version retrieval in conf.py and enhance version consistency test

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

saicheranb committed:
Fix version consistency check in test_version_is_consistent

saicheranb committed:
Fix version consistency check in test_version_is_consistent function

saicheranb committed:
Fix version consistency check in test_version_is_consistent to handle dynamic versioning

pre-commit-ci[bot] committed:
[pre-commit.ci] auto fixes from pre-commit.com hooks

for more information, see https://pre-commit.ci

saicheranb committed:
Update Sphinx dependency to allow any version in pyproject.toml

saicheranb committed:
Update Sphinx dependency to version range >=5.3,<6.0 in pyproject.toml

sebhmg commented on pyproject.toml:
```diff
@@ -116,6 +106,86 @@ initial-content = """
+pyyaml = '*'
+jinja2 = '*'
+packaging = '*'
+sphinx = ">=5.3,<6.0"
```
**suggestion:** move sphinx to a doc feature

and
1. change test task to run pytest on `tests` folder only
2. add a test task in a test-doc environment that runs pytest on docs only
3. define a test-doc environment with features: py310, test, doc

OK not to run test-doc in CI for now: we do not publish that doc to RTD

sebhmg replied:
**suggestion:** simpler

```suggestion
sphinx = ">=5.3, <5.*"
```

saicheranb committed:
Refine dependency specifications in pyproject.toml

- Updated numpy dependency to use exact versioning (1.26.*).
- Changed pypng and vectormath dependencies to specific versions (0.20220715.* and 0.2.* respectively).
- Adjusted six dependency to allow only version 1.*.
- Modified sphinx dependency to specify version range (>=5.3, 5.*).
- Added new test-doc task for documentation testing.
- Enhanced test command to include tests directory.

saicheranb committed:
Add testpaths configuration for pytest in pyproject.toml

saicheranb committed:
Refactor test documentation task configuration in pyproject.toml

saicheranb committed:
Define test-doc task with enforced Pixi environment for documentation testing via pytest docs.

sebhmg committed:
[DEVOPS-693] re-apply recommendations by pixi for lock file

sebhmg committed:
[DEVOPS-693] remove irrelevant entries from gitattribute

sebhmg committed:
[DEVOPS-693] allow commit of .pixi/config.toml if any

sebhmg committed:
[DEVOPS-693] finalize changes

- re-organize to keep everything together for a given feature
- define task test-docs to autoselect the environment
- add inderect deps so they they do not come from pypi
- add future geoh5py version in project: ok if it cannot be installed for now
- add installation of package for doc (or it fails)

sebhmg committed:
[DEVOPS-693] fix indirect dependency versions

As indirect dependencies come through pip, and we force them from conda,
their versions must be specified, or they will conflict with the most recent versions from conda

sebhmg committed:
[DEVOPS-693] add more keywords in package meta info

sebhmg committed:
[DEVOPS-693] aesthetics: reorder trove classifiers

sebhmg committed:
[DEVOPS-693] also resolve doc environment with solve-group default

sebhmg committed:
[DEVOPS-693] relock pixi

sebhmg committed:
[DEVOPS-693] poetry check with plugins works if not check poetry.lock

sebhmg committed:
[DEVOPS-693] move pixi config to its own file

less risk to break python.toml because error in large config

also move ipython to dev dependencies

sebhmg committed:
[DEVOPS-693] aesthetics

sebhmg committed:
[DEVOPS-693] add keywords and classifiers troves

sebhmg committed:
[DEVOPS-693] pixi lock again

sebhmg committed:
[DEVOPS-693] use pixi to run pylint from pre-commit
